zwave_js_server/util/lock.py

```diff
@@ -144,3 +151,58 @@ async def clear_usercode(node: Node, code_slot: int) -> SetValueResult | None:
     """Clear a code slot on the lock."""
     value = get_code_slot_value(node, code_slot, LOCK_USERCODE_STATUS_PROPERTY)
     return await node.async_set_value(value, CodeSlotStatus.AVAILABLE.value)
+
+
+async def set_configuration(
+    endpoint: Endpoint, configuration: DoorLockCCConfigurationSetOptions
+) -> SupervisionResult | None:
+    """Set lock configuration."""
+    # It is invalid to set the operation to timed with no timeout, or to constant
+    # with a timeout
+    if configuration.operation_type == (
+        OperationType.TIMED ^ (configuration.lock_timeout_configuration is None)
+    ):
+        raise ValueError(
+            "Invalid operation type and lock timeout configuration combination"
+        )
+    errors: list[str] = []
+
+    for property_name, attr_name in (
+        (CURRENT_AUTO_RELOCK_TIME_PROPERTY, "auto_relock_time"),
+        (CURRENT_HOLD_AND_RELEASE_TIME_PROPERTY, "hold_and_release_time"),
+        (CURRENT_TWIST_ASSIST_PROPERTY, "twist_assist"),
+        (CURRENT_BLOCK_TO_BLOCK_PROPERTY, "block_to_block"),
+    ):
+        # It a value for a particular configuration value is not provided and it exists
+        # on the node, use the cached value
+        cached_value = next(
+            (
+                value
+                for value in endpoint.values.values()
+                if value.command_class == CommandClass.DOOR_LOCK
+                and value.property_name == property_name
+            ),
+            None,
+        )
+        if (
+            val := getattr(configuration, attr_name)
+        ) is not None and cached_value is None:
```

Comment:
Start with the check of `cached_value` as that will be the fastest. We'll then avoid the `getattr` if the first check is false.

Reply:
but we need to know the `val` in the next check, and we don't want to retrieve it twice right?

Reply:
Ok